src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp

```diff
@@ -0,0 +1,690 @@
+    const auto& GemmABlockCopyClusterLengths_GemmM = WeiBlockCopyClusterLengths_GemmM;
+
+    if(!(GemmKPerBlock % GemmBBlockCopyClusterLengths_GemmK == 0 &&
+         GemmKPerBlock % GemmABlockCopyClusterLengths_GemmK == 0 &&
+         GemmNPerBlock % GemmBBlockCopyClusterLengths_GemmN == 0 &&
+         GemmMPerBlock % GemmABlockCopyClusterLengths_GemmM == 0))
+        return false;
+
+    // unsupported xdlops-gemm
+    if(GemmMPerWave == 16 && GemmNPerWave == 32)
+        return false;
+    if(GemmMPerWave == 32 && GemmNPerWave == 16)
+        return false;
+    if(GemmMPerWave == 8 && GemmNPerWave != 64)
+        return false;
+    if(GemmMPerWave == 4 && GemmNPerWave != 64)
+        return false;
+
+    const auto WaveSize  = 64;
+    const auto BlockSize = GemmNPerBlock * GemmMPerBlock / (GemmMPerWave * GemmNPerWave) * WaveSize;
+
+    // fail with blockSize >= 512
+    /// \todo fix the issue with blockSize >= 512
+    if(BlockSize < 64 || BlockSize > 256)
+        return false;
+
+    if(BlockSize != GemmBBlockCopyClusterLengths_GemmK * GemmBBlockCopyClusterLengths_GemmN)
+        return false;
+
+    if(BlockSize != GemmABlockCopyClusterLengths_GemmM * GemmABlockCopyClusterLengths_GemmK)
+        return false;
+
+    if((GemmMPerBlock % GemmMPerWave) != 0 || (GemmNPerBlock % GemmNPerWave) != 0)
+        return false;
+
+    const auto GemmBBlockCopyThreadSliceLengths_GemmN =
+        GemmNPerBlock / GemmBBlockCopyClusterLengths_GemmN;
+    const auto GemmABlockCopyThreadSliceLengths_GemmM =
+        GemmMPerBlock / GemmABlockCopyClusterLengths_GemmM;
+    const auto lds_size = ComputeLDSRequiredSize(ctx,
+                                                 GemmNPerBlock,
+                                                 GemmMPerBlock,
+                                                 GemmKPerBlock,
+                                                 1,
+                                                 1,
+                                                 GemmBBlockCopyThreadSliceLengths_GemmN,
+                                                 GemmABlockCopyThreadSliceLengths_GemmM,
+                                                 GetEPackLength(ctx, true));
+    return lds_size <= 64 * 1024;
+}
+
+PerformanceImplicitGemmBwdDataV4R1Xdlops::PerformanceImplicitGemmBwdDataV4R1Xdlops(bool spare)
+{
+    GemmNPerBlock = spare ? 16 : 64;
+    GemmMPerBlock = spare ? 4 : 64;
+    GemmKPerBlock = spare ? 4 : 8;
+
+    GemmMPerWave = spare ? 4 : 64;
+    GemmNPerWave = spare ? 16 : 64;
+
+    InBlockCopyClusterLengths_GemmK = 4;
+    InBlockCopyClusterLengths_GemmN = 4;
+
+    WeiBlockCopyClusterLengths_GemmK = 2;
+    WeiBlockCopyClusterLengths_GemmM = 4;
+
+    use_spare_set = spare;
+}
+
+PerformanceImplicitGemmBwdDataV4R1Xdlops::PerformanceImplicitGemmBwdDataV4R1Xdlops(
+    int BPerBlock_,
+    int KPerBlock_,
+    int EPerBlock_,
+    int GemmMPerWave_,
+    int GemmNPerWave_,
+    int InBlockCopyClusterLengths_E_,
```

Comment:
Please rename from InBlock/WeiBlock to GemmN/GemmM

Reply:
Renamed.